modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/PlanUtils.java

```diff
@@ -144,14 +144,33 @@ public static Mapping computeAggFieldMapping(List<ImmutableBitSet> groupingSets)
             }
         }
 
-        Mapping mapping = Mappings.create(MappingType.INVERSE_SURJECTION, fieldIndices.length(), fieldIndices.cardinality());
+        return sortedValuesIndexMapping(fieldIndices);
+    }
+
+    /**
+     * Creates a mapping such that each value is replaced by its position in the ordered list.
+     *
+     * <p>For example, for the following values {@code [2, 3, 5, 7, 9]} the sorting positions
+     * will be {@code [0, 1, 2, 3, 4]}, the resulting mapping will be:
+     * <pre>
+     *  2 -> 0
+     *  3 -> 1
+     *  5 -> 2
+     *  7 -> 3
+     *  9 -> 4
+     * </pre>
+     *
+     * @param values Values to be mapped to its position in the ordered list.
+     */
+    public static Mapping sortedValuesIndexMapping(BitSet values) {
```

Comment:
well, we already have `org.apache.ignite.internal.sql.engine.util.Commons#trimmingMapping`...